apps/verification/src/verification.service.ts

```diff
@@ -187,17 +187,17 @@ export class VerificationService {
       return getProofPresentationById?.response;
     } catch (error) {
       this.logger.error(`[getProofPresentationById] - error in get proof presentation by proofId : ${JSON.stringify(error)}`);
-      const errorStack = error?.response?.error?.reason;
+      const errorMessage = error?.response?.error?.reason || error?.message;
 
-      if (errorStack) {
-        throw new RpcException({
-          message: ResponseMessages.verification.error.proofNotFound,
-          statusCode: error?.response?.status,
-          error: errorStack
+      if (errorMessage?.includes('record with id') && errorMessage?.includes('not found')) {
+        throw new NotFoundException({
+          statusCode: 404,
+          message: errorMessage,
+          error: 'Not found'
         });
```

Comment:
```suggestion
        throw new NotFoundException(errorMessage);
```

I think this should be enough as the status code is handled internally